src/Hazelcast.Net/Networking/CloudDiscovery.cs

```diff
@@ -68,8 +68,12 @@ internal static void SetResponse(string response)
 
             try
             {
-                // TODO: is this the best way to do an http request?
+                // TODO: Make it async and consider to use HttpClientFactory
+#pragma warning disable SYSLIB0014
```

Comment:
Ideally we want to have this code (maybe you can add it but comment it out):
```
string response;
using (var httpClient = new HttpClient())
using (var httpRequest = new HttpRequestMessage(HttpMethod.Get, _endpointUrl))
{
    httpClient.Timeout = TimeSpan.FromMinutes(_connectionTimeoutMilliseconds);
    httpRequest.Headers.AcceptCharset.ParseAdd("UTF-8");
    response = await httpClient.GetStringAsync(_endpointUrl).CfAwait();
}
```
problem as you have noted is that this forces us to make `CloudDiscovery.Scan()` async which eventually bubbles up to making some of `AddressProvider` methods async... so let's not do it now but maybe be more explicit in the TODO?

Reply:
Yeap, I didn't want to mix the removing warnings and the refactoring, so added a better explanation at https://github.com/hazelcast/hazelcast-csharp-client/pull/682/commits/146cc487e6723800e0609781507b11b70e1c1c41

Reply:
ok